### PR TITLE
Restructure the /pages directory and reorganize documentation

### PR DESCRIPTION
**Pull request by eddietejeda**
Previously, when you looked in the pages/ folder, you'd notice a couple of markdown pages and a couple sub-folders with additional markdown pages. This folder structure mapped directly to the URL path. But as I started reorganizing the documentation, the permalinks stopped matching the folder structure, so I flattened the structure.

Proposed changes in this pull request:
- Eliminate the folder structure and instead have all the pages/ in one flat folder. I then use the permalink front matter option to specify the URL.
- This might seem messier, but in reality it's much clearer. The previous folder structure meant that if you wanted to find a file you have to hunt through different folders and sometimes the content did not match the folder name
- We don't have that many pages to begin with, so eliminating the folders does not cause too much clutter in the pages/ path.
- I also took this opportunity to pull out "Support", which was buried in the documentation, and created a top level menu item for it.

Potential Impact
- I made sure to have a redirect from the old URL path to the new, but we should make sure that there are no dead links 
- This is a big change and we should make sure we test it properly before releasing to production.

/cc @amirbey @davemcorwin 

[CircleCI](https://circleci.com/gh/18F/federalist.18f.gov/tree/restructure_pages_directory)

[:sunglasses: PREVIEW](https://federalist-proxy.app.cloud.gov/preview/18f/federalist.18f.gov/restructure_pages_directory/)

### CODE REVIEW

File: pages/documentation/index.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 ---
 title: Using Federalist
-permalink: /pages/using-federalist/
+permalink: /documentation
```
Comment (amirbey) [inline pages/documentation/index.md]: missing ending slash?

Comment (eddietejeda) [reply pages/documentation/index.md]: done

**Comment by davemcorwin** [comment]
`/documentation/` 404s...

File: _config.yml [hunk 2]
```diff
@@ -20,21 +20,21 @@ search_site_handle: federalist
...
-    values:
-      layout: "page"
-      sidenav: "sidenav"
-  - scope:
-      path: ""
-      type: templates
-    values:
-      layout: "default"
+    permalink: /:path/
+#
```
Comment (davemcorwin) [inline _config.yml]: delete these instead of just commenting them out

**Comment by eddietejeda** [comment]
@davemcorwin I fixed the 404 issue. Can you test again?

**Comment by davemcorwin** [comment]
@eddietejeda more 404s:
- `/how-federalist-works/node-on-federalist` FROM `/documentation/supported-site-engines/`
- `/documentation/support/` FROM `/documentation/content-guide/`

**Comment by eddietejeda** [comment]
Hey @davemcorwin and @amirbey.

I added a link checker to the build [process](https://github.com/18F/federalist.18f.gov/pull/289/files#diff-1d37e48f9ceff6d8030570cd36286a61) 

All tests are now passing. I think we are ready to merge. Please review.

**Comment by amirbey** [comment]
renders w/o layout -https://federalist-proxy.app.cloud.gov/preview/18f/federalist.18f.gov/restructure_pages_directory/documentation/monorepos-on-federalist/

Screen Shot 2019-03-27 at 4 11 56 PM

**Pull request merged**